src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/retryfailedmanagedindex/TransportRetryFailedManagedIndexAction.kt

```diff
@@ -70,9 +70,10 @@ class TransportRetryFailedManagedIndexAction @Inject constructor(
         private val request: RetryFailedManagedIndexRequest
     ) {
         private val failedIndices: MutableList<FailedIndex> = mutableListOf()
-        private val listOfIndexMetaDataBulk: MutableList<ManagedIndexMetaData> = mutableListOf()
-        private val listOfIndexMetaData: MutableList<Pair<Index, ManagedIndexMetaData>> = mutableListOf()
-        private var updated: Int = 0
+        private val listOfMetadata: MutableList<ManagedIndexMetaData> = mutableListOf()
+        private val listOfIndexToMetadata: MutableList<Pair<Index, ManagedIndexMetaData>> = mutableListOf()
+        private val mapOfItemIdToIndex: MutableMap<Int, Index> = mutableMapOf()
+        private lateinit var clusterState: ClusterState
```

Comment:
Generic comment for whole file:

What happens when you have ManagedIndexMetadata from the cluster state and ManagedIndexMetadata from the config index (i.e. the migration started but failed on delete part)?

Also what happens when only cluster state has the ManagedIndexMetadata (i.e. migration hasn't happened yet)?

Reply:
now using metadataservice to move metadata, ISM only rely on metadata saved in/moved to config index, if metadata stay in cluster state, ISM will wait until it get moved to config index